src/resources/MachineLearning/CCConfiguration/CCConfigurationInterfaces.ts

```diff
@@ -0,0 +1,33 @@
+export enum Operator {
+    Equals = 'EQUALS',
+    NotEquals = 'NOT_EQUALS',
+}
+
+export interface FilterConditions {
+    field: string;
+    operator: Operator;
+    value: string;
+}
+
+export interface CCConfigurationModelBase {
+    modelId: string;
+    modelDisplayName: string;
+    sources: string[];
+    caseFilterConditions?: FilterConditions[];
+    caseIdField: string;
+    fieldsForModelTraining: string[];
+    fieldsToPredict: string[];
+}
+
+export interface FixedExtractionPeriod {
+    startTime: number;
+    endTime: number;
+}
+
+export interface IntervalExtractionPeriod {
+    exportPeriod: string;
+}
+
+export type ExtractionPeriod = FixedExtractionPeriod | IntervalExtractionPeriod;
+
+export type CCConfigurationModel = CCConfigurationModelBase & ExtractionPeriod;
```

Comment:
Ok, I have litteraly no prior knowledge of .ts, so here come's a noob question.
The actual format is 
```
{
   "field":"color",
   "documentExtractionPeriod":{
      "exportPeriod":"P1Y"
   },
   "sources":[
      "mySource1",
      "mySource2"
   ],
   "caseFilterConditions":[
      {
         "field":"iscasedclosed",
         "operator":"==",
         "Value":"true"
      }
   ]
}
```
As you can see, the `exportPediod`, vs `startTime` and `endTime` is expected to be nested under `documentExtractionPeriod`. Should `CCConfigurationModel` reflect that, or will that be handled somewhere else?

Reply:
Oh i missed that. You are correct. I made a mistake here.

Reply:
i thought you said the operator values had to be "EQUALS" or "NOT_EQUALS" though...?

Reply:
Yes, my bad, it's the old thing :sweat_smile: the conditions is still "EQUALS" or "NOT_EQUALS".